src/agent/agent.c

```diff
@@ -1043,6 +1044,19 @@ static int agent_method_subscribe(sd_bus_message *m, void *userdata, UNUSED sd_b
                 return sd_bus_reply_method_errorf(m, SD_BUS_ERROR_INVALID_ARGS, "Invalid arguments");
         }
 
+        if (is_wildcard(unit)) {
+                if (agent->wildcard_subscription_active) {
+                        return sd_bus_reply_method_errorf(
+                                        m, SD_BUS_ERROR_FAILED, "Already wildcard subscribed");
+                }
+                agent->wildcard_subscription_active = true;
+
+                AgentUnitInfo info = { NULL, (char *) unit, true, true, UNIT_ACTIVE, NULL };
```

Comment:
UNIT_ACTIVE for a "*" unit seems a bit weird to me, but unfortunately I don't really see a better alternative, and I don't think we want to add new possible states.

Reply:
I think we should have a substate though.

Reply:
Yes, I agree that the `UnitNew` doesn't really fit. The unit new signal only contains the name of the unit and the reason - so every other information like the active state is not used. `AgentInfo` is only created for the sake of calling the `agent_emit_unit_new` function. Maybe replacing `UNIT_ACTIVE` with a -1 is more clear?

Reply:
_UNIT_ACTIVE_STATE_INVALID maybe? Although I don't really think it matters. Lets leave it.